File: java_11_Interfaces/src/classes/Bird.java
package classes;

import interfaces.Flyable;
import interfaces.Swimmable;

public class Bird implements Swimmable, Flyable {
    String name;
    int age;

    public Bird(String name, int age) {
        this.name = name;
        this.age = age;
    }

    @Override
    public void swim() {
        System.out.println(name + " плавает болтая ножками");
    }

    @Override
    public void fly() {
        System.out.println(name + " летит взмахивая крыльями уже " + age + " лет");
    }
}

File: kotlin_19_OverrideFunctionsAndProperties/src/Bird.kt
import kotlin.random.Random

open class Bird(
    var type: String,
    open var flyable: Boolean = true,
    open var swimmable: Boolean = true,
) {
    fun birdInfo() {
        println("Птица относящаяся к типу $type, ${able(flyable)} летать, ${able(swimmable)} плавать")
    }

    open fun move() {
        when {
            (flyable and swimmable) -> if (Random.nextBoolean()) fly() else swim()
            (flyable and !swimmable) -> fly()
            (!flyable and swimmable) -> swim()
        }
    }

    private fun fly() {
        println("Птица полетела")
    }

    private fun swim() {
        println("Птица поплыла")
    }

    private fun able(able: Boolean) = if (able) "умеет" else "не умеет"
}